src/mapping-v3-core.ts

```diff
@@ -93,7 +106,8 @@ function _handleMint<T>(contract: T, event: Mint): void {
   if (
```

Comment:
I think every section before these type of checks needs an additional check:
```
const engineFlexContract = getV3EngineFlexContract(event.address);
  if (engineFlexContract) {
    _handleMint(engineContract, event);
    return;
  }
```

Reply:
ah roger, but I think this can just be added once to the upper level `handleMint` function if im understanding correctly

Reply:
yes ser! sorry, Github being a bit weird about where I could place this comment 😓

Reply:
I think the `tokenIdToHash` call is the only thing that's preventing us from having a single handler here that uses `IGenArt721CoreContractV3_Base.bind(contractAddress)` for everything so that we don't need to do the whole contract type check pattern. I think that's common across all of the V3 contracts so it would make sense to me to add to the interface.

I'd imagine a lot of the other handlers for events exported from `../generated/IGenArt721CoreV3_Base/IGenArt721CoreContractV3_Base` could be updated similarly?

Reply:
@yoshiwarab ah, yes, good call, agree for the case of this `handleMint` handler! This might also be true for a few of the other handlers, and seems best to rely on `IGenArt721CoreContractV3_Base.bind` where possible and straightforward. For things like `handlePlatformUpdated` I _think_ the current pattern of binding to the contract type is just clean because it enables statements like `if (contract instanceof GenArt721CoreV3_Engine)` in areas where the handling behaviors diverge between the engine and flagship (e.g. when dealing with payment addresses or indexing of minter suite).

Reply:
Cool, yeah makes sense! Also, I think this more general refactor should be a follow on rather than part of this PR

Reply:
@lyaunzbe - I'm happy to pick up this refactoring work in a fast-follow PR if that sounds good to you? 👍

Reply:
@ryley-o yes, that would be much appreciated! thank you ser 🙏 🙏